pkgs/tools/graphics/gmic/default.nix

```diff
@@ -70,48 +62,60 @@ stdenv.mkDerivation rec {
     "-DUSE_SYSTEM_CIMG=ON"
   ];
 
-  postPatch = ''
-    # TODO: build from source
+  postPatch = let
+    # TODO: build this from source
+    # Reference: src/Makefile, directive gmic_stdlib.h
+    gmic_stdlib = fetchurl {
+      name = "gmic_stdlib.h";
+      url = "http://gmic.eu/gmic_stdlib${lib.replaceStrings ["."] [""] finalAttrs.version}.h";
+      hash = "sha256-ExMCxFkkctqrdSy5M/TXD5GBRmRA9YEdsYW8nWiTEYY=";
+    };
+  in ''
```

Comment:
This will make it more difficult to override because overriding `version` still requires this hash on `gmic_stdlib` to be changed (and before `gmic.overrideAttrs (oldAttrs: { version = ...; src = ...; gmic_stdlib = ...; })` would have been an option)

Maybe the derivation should just use `srcs` instead?

Reply:
You are right. I will "revert" this.

Well, we can investigate how this file can be generated from sources to begin with. Reading the Makefile, it looks like this file can be generated by calling another directive.

Reply:
Oh frak, this is self-referent! It uses gmic to generate gmic-stdlib!

https://github.com/GreycLab/gmic/blob/cc41e1d5de0fd01f6ae48d5d8df6fc73a9188558/src/Makefile#L735

I need to re-read all theory about bootstrap in Nixpkgs...

Reply:
> Oh frak, this is self-referent! It uses gmic to generate gmic-stdlib!

Yep, I noticed that too when I last looked at it...

We only started needing to download it separately when we started needing to pull the main sources from GitHub (since the tarball actually has the file already in it). We had to pull from GitHub since the CMake files stopped being included with the tarball a few versions back (#179646)

We could perhaps switch to the Makefile-based G'MIC build which is included in the tarball, but I'm not sure we can get G'MIC-Qt to find that properly without the generated CMake file...

It's a bit of a mess tbh so downloading is probably best for now

Reply:
Also I think it may have broken the updateScript when it wasn't an attr on the derivation anymore, but I'd have to double check to be sure

Either way, just pulling it back out right now looks good, thanks!